### PR TITLE
fix: Add retry logic to auth0 calls

### PR DESCRIPTION
**Pull request by Bento007**
- #2980

### Reviewers
**Functional:** @danieljhegeman 

**Readability:** @ebezzi 

---

## Changes
- add retry logic for calls to auth0

### CODE REVIEW

File: backend/corpora/common/authorizer.py [hunk 1]
```diff
@@ -2,13 +2,20 @@
...
+from requests.packages.urllib3.util import Retry
 from jose import jwt
 from jose.exceptions import ExpiredSignatureError, JWTError, JWTClaimsError
 
 from .corpora_config import CorporaAuthConfig
 from backend.corpora.common.utils.http_exceptions import UnauthorizedError
 
 
+auth0_session_with_retry = requests.Session()
+retry_config = Retry(total=3, backoff_factor=1, status_forcelist=[429, 500, 502, 503, 504])
```
Comment (danieljhegeman) [inline backend/corpora/common/authorizer.py:15]: why are we specifying `status_forcelist` twice? Both here and in the `retry_config`

**Review by danieljhegeman — COMMENTED** [review]
One question: it seems like we're specifying the status forcelist twice when we (probably?) don't need to be; could lead to mistakes when adding/removing statuses later

File: backend/corpora/common/authorizer.py (hunk 1 above)
Comment (Bento007) [reply backend/corpora/common/authorizer.py:15]: I can consolidate

File: backend/corpora/common/auth0_manager.py [hunk 2]
```diff
@@ -12,12 +15,20 @@ class Auth0ManagementSession:
 
     _session = None
     _domain = None
+    config = CorporaAuthConfig()
```
Comment (danieljhegeman) [inline backend/corpora/common/auth0_manager.py:18]: 👍

**Review by danieljhegeman — APPROVED** [review]
Looks good, but random q: do we have any hard evidence that the Auth0 requests were timing out? Or is this just our best guess at what was causing transient failures, and at worst it's simply precautionary?

**Comment by Bento007** [comment]
This is our best guess. I have a separate PR to log 500 errors from the server.

**Pull request merged**